File: PHPMODELS/Product.php
<?php

/**
 * Created by Reliese Model.
 */

namespace App\Models;

use Carbon\Carbon;
use Illuminate\Database\Eloquent\Collection;
use Illuminate\Database\Eloquent\Model;

/**
 * Class Product
 * 
 * @property int $id
 * @property string $name
 * @property string|null $description
 * @property float|null $price
 * @property float $cost_price
 * @property int $category_id
 * @property bool $isEnabled
 * @property int $sale_count
 * @property int $review_count
 * @property string|null $sku
 * @property int|null $stock
 * @property string|null $observation
 * @property float|null $weight
 * @property float|null $height
 * @property float|null $width
 * @property float|null $length
 * @property string|null $product_tiny
 * 
 * @property Category $category
 * @property Collection|ProductImage[] $product_images
 * @property Collection|ProductOrderItem[] $product_order_items
 * @property Collection|PropertyValue[] $property_values
 * @property Collection|Review[] $reviews
 * @property Collection|StoreFeatured[] $store_featureds
 * @property Collection|StoreShippingOrderItem[] $store_shipping_order_items
 *
 * @package App\Models
 */
class Product extends Model
{
	protected $table = 'products';
	// public $timestamps = false;

	protected $casts = [
		'price' => 'float',
		'cost_price' => 'float',
		'category_id' => 'int',
		'isEnabled' => 'bool',
		'isPreSale' => 'bool',
		'sale_count' => 'int',
		'review_count' => 'int',
		'stock' => 'int',
		'weight' => 'float',
		'height' => 'float',
		'width' => 'float',
		'length' => 'float'
	];

	protected $fillable = [
		'name',
		'description',
		'price',
		'cost_price',
		'category_id',
		'isEnabled',
		'isPreSale',
		'sale_count',
		'review_count',
		'sku',
		'stock',
		'observation',
		'weight',
		'height',
		'width',
		'length',
		'product_tiny',
		'ncm',
		'ean',
		'marca',
		'cest',
		'stop_stock',
		'promotion_id',
		'original_price'
	];

	public function category()
	{
		return $this->belongsTo(Category::class);
	}

	public function product_images()
	{
		return $this->hasMany(ProductImage::class);
	}

	public function product_users()
	{
		return $this->hasMany(ProductUser::class);
	}

	public function product_order_items()
	{
		return $this->hasMany(ProductOrderItem::class);
	}

	public function property_values()
	{
		return $this->hasMany(PropertyValue::class);
	}

	public function reviews()
	{
		return $this->hasMany(Review::class);
	}

	public function store_featureds()
	{
		return $this->hasMany(StoreFeatured::class);
	}

	public function store_shipping_order_items()
	{
		return $this->hasMany(StoreShippingOrderItem::class);
	}

	public function product_promotions()
	{
		return $this->hasMany(ProductPromotions::class);
	}

	public function product_pre_sale()
	{
		return $this->hasOne(PreSaleProducts::class);
	}
}

File: internal/comum/models/product.go
package models

import (
	"database/sql"
	"time"
)

// Product representa um produto no sistema
type Product struct {
	ID            int             `db:"id" json:"id"`
	Name          string          `db:"name" json:"name"`
	Description   sql.NullString  `db:"description" json:"description,omitempty"`
	Price         sql.NullFloat64 `db:"price" json:"price,omitempty"`
	CostPrice     float64         `db:"cost_price" json:"cost_price"`
	CategoryID    int             `db:"category_id" json:"category_id"`
	IsEnabled     bool            `db:"isEnabled" json:"isEnabled"`
	IsPreSale     bool            `db:"isPreSale" json:"isPreSale"`
	SaleCount     sql.NullInt64   `db:"sale_count" json:"sale_count,omitempty"`
	ReviewCount   sql.NullInt64   `db:"review_count" json:"review_count,omitempty"`
	SKU           sql.NullString  `db:"sku" json:"sku,omitempty"`
	Stock         sql.NullInt64   `db:"stock" json:"stock,omitempty"`
	Observation   sql.NullString  `db:"observation" json:"observation,omitempty"`
	Weight        sql.NullFloat64 `db:"weight" json:"weight,omitempty"`
	Height        sql.NullFloat64 `db:"height" json:"height,omitempty"`
	Width         sql.NullFloat64 `db:"width" json:"width,omitempty"`
	Length        sql.NullFloat64 `db:"length" json:"length,omitempty"`
	ProductTiny   sql.NullString  `db:"product_tiny" json:"product_tiny,omitempty"`
	NCM           sql.NullString  `db:"ncm" json:"ncm,omitempty"`
	EAN           sql.NullString  `db:"ean" json:"ean,omitempty"`
	Marca         sql.NullString  `db:"marca" json:"marca,omitempty"`
	CEST          sql.NullString  `db:"cest" json:"cest,omitempty"`
	StopStock     sql.NullInt64   `db:"stop_stock" json:"stop_stock,omitempty"`
	PromotionID   sql.NullInt64   `db:"promotion_id" json:"promotion_id,omitempty"`
	OriginalPrice sql.NullFloat64 `db:"original_price" json:"original_price,omitempty"`
	CreatedAt     time.Time       `db:"created_at" json:"created_at"`
	UpdatedAt     time.Time       `db:"updated_at" json:"updated_at"`
}

// TableName retorna o nome da tabela no banco de dados
func (Product) TableName() string {
	return "products"
}
